server/ai_http.go

```diff
@@ -271,6 +274,29 @@ func (h *lphttp) StartLiveVideoToVideo() http.Handler {
 			return
 		}
 
+		// Precreate the channels to avoid race conditions
+		// TODO get the expected mime type from the request
+		pubCh := trickle.NewLocalPublisher(h.trickleSrv, mid, "video/MP2T")
```

Comment:
Do I understand correctly that these 4 lines are not really needed, because the channel will be anyway created when requested from Gateway. However, we create them here, just because we want to subscribe them below and we don't want to have some race condition when subscribing to the non-existing channel? Is that correct?

Reply:
They are needed, because of race conditions:

* gateway may subscribe before the job sends anything into the output stream
* job may start subscribing before the gateway starts sending the input (as well as the O wanting to subscribe to the input itself)